tests/tt_metal/tt_metal/perf_microbenchmark/ethernet/test_ethernet_read_and_send_data.cpp

```diff
@@ -0,0 +1,357 @@
+        "Sending {} bytes from device {} eth core {} addr {} to device {} eth core {} addr {}",
+        input_buffer_size_bytes,
+        sender_device->id(),
+        eth_sender_core.str(),
+        src_eth_l1_byte_address,
+        receiver_device->id(),
+        eth_receiver_core.str(),
+        dst_eth_l1_byte_address);
+    std::cout << "num_messages_to_send: " << num_messages_to_send << std::endl;
+    // Generate inputs
+    ////////////////////////////////////////////////////////////////////////////
+    //   SETUP THE INPUT CB
+    ////////////////////////////////////////////////////////////////////////////
+    std::cout << "Generating vector" << std::endl;
+    auto inputs = generate_uniform_random_vector<uint32_t>(0, 100, input_buffer_size_bytes / sizeof(uint32_t));
+
+    // Clear expected value at ethernet L1 address
+
+    BankedConfig test_config =
+        BankedConfig{
+            .num_pages = num_pages,
+            .size_bytes = input_buffer_size_bytes,
+            .page_size_bytes = input_buffer_page_size,
+            .input_buffer_type = source_is_dram ? BufferType::DRAM : BufferType::L1,
+            .output_buffer_type = dest_is_dram ? BufferType::DRAM : BufferType::L1,
+            .l1_data_format = tt::DataFormat::Float16_b};
+    auto input_buffer =
+            CreateBuffer(
+                InterleavedBufferConfig{sender_device, test_config.size_bytes, test_config.page_size_bytes, test_config.input_buffer_type});
+
+    bool input_is_dram = test_config.input_buffer_type == BufferType::DRAM;
+    tt_metal::detail::WriteToBuffer(input_buffer, inputs);
+    const uint32_t dram_input_buf_base_addr = input_buffer->address();
+
+    ////////////////////////////////////////////////////////////////////////////
+    //   EMPTY INITIALIZE THE OUTPUT CB
+    ////////////////////////////////////////////////////////////////////////////
+
+    // Clear expected value at ethernet L1 address
+    std::vector<uint32_t> all_zeros(inputs.size(), 0);
+
+    auto output_buffer =
+            CreateBuffer(
+                InterleavedBufferConfig{receiver_device, test_config.size_bytes, test_config.page_size_bytes, test_config.output_buffer_type});
+
+    bool output_is_dram = test_config.output_buffer_type == BufferType::DRAM;
+    tt_metal::detail::WriteToBuffer(output_buffer, all_zeros);
+    const uint32_t dram_output_buffer_base_addr = output_buffer->address();
+
+
+
+    ////////////////////////////////////////////////////////////////////////////
+    //                      Sender Device
+    ////////////////////////////////////////////////////////////////////////////
+    tt_metal::Program sender_program = tt_metal::Program();
+
+    uint32_t num_pages_per_l1_buffer = num_bytes_per_send / input_buffer_page_size;
+    TT_ASSERT(num_messages_to_send * num_pages_per_l1_buffer >= num_pages);
+    std::cout << "eth_l1_mem::address_map::ERISC_L1_UNRESERVED_BASE: " << eth_l1_mem::address_map::ERISC_L1_UNRESERVED_BASE << std::endl;
```

Comment:
Total unreserved L1 is will be bigger after tunneling changes, double check that things are still passing

Reply:
Current plan is to rebase to main after tunneling is in and re-test everything.